walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -26,6 +26,19 @@
    * the request ensuring that the attandees don't have any conflicts with other events.
    */
   public Collection<TimeRange> query(Collection<Event> events, MeetingRequest request) {
+    if (!request.getOptionalAttendees().isEmpty()) {
+      List<TimeRange> requestAttendeeRanges = filterToRequestAttendeeTimeRangesOptional(
+          events, request.getAttendees(), request.getOptionalAttendees());
+      Collections.sort(requestAttendeeRanges, TimeRange.ORDER_BY_START);
+      requestAttendeeRanges = combineOverlaps(requestAttendeeRanges);
+      List<TimeRange> result =
+          findMeetingRangesWithNoConflict(requestAttendeeRanges, request.getDuration());
+      if (!result.isEmpty()) {
+        return result;
+      } else if (request.getAttendees().isEmpty()) {
+        return Collections.emptyList();
+      }
+    }
     List<TimeRange> requestAttendeeRanges =
```

Comment:
Should this part belong to an 'else' condition? Moving to an else condition will make the logic clearer especially when you reuse the variable name.

Also I think the logics are similar, except for the call to get the initial requestAttendeeRanges. Can you extract the common logics outside the if-else condition? Looks like it can be written as:

List<TimeRange> requestAttendeeRanges = request.getOptionalAttendees().isEmpty()
    ? filterToRequestAttendeeTimeRanges(events, request.getAttendees())
    : filterToRequestAttendeeTimeRangesOptional(events, request.getAttendees(), request.getOptionalAttendees());

Collections.sort(requestAttendeeRanges, TimeRange.ORDER_BY_START);
requestAttendeeRanges = combineOverlaps(requestAttendeeRanges);

...

Reply:
There is a case where the request has optional attendees and mandatory attendees with no time range where all of them can attend. For this case, I would first check to see if there is a time where all the optional and mandatory attendees can attend. If not, I would use the logic underneath to see if there is a time where all the mandatory attendees can attend. 

Putting the logic below in an else statement would mean that it wouldn't be executed in this case since the optional attendee list isn't empty. 

Also I need requestAttendeeRanges to be reassigned a value based on the result from the logic in the if statement. I will rename the variable to make it more clear.